typing_tests/pine-params.ts

```diff
@@ -250,7 +250,7 @@ await (async () => {
 		options: {
 			$select: ['id', 'device_name'],
 		},
-	});
+	} as const);
```

Comment:
I expect that this is going to be an intrusive change for TS consumers to do in a patch :(
Especially since I don't expect adding `as const` to be obvious to users.
I will try to find a workaround.

Reply:
Honestly I'm not sure it's worth trying to fix @thgreasi, I think the simple fact might be that in order to make these typings work without breaking changes just might not be possible since I think we might well hit an issue where it relies on bugs that force an any and it is not fixable without `as const` (or some brand new version of typescript or otherwise some change that is not actually backwards compatible. So overall I think it'd make more sense to just eat the major and switch to using the `pinejs-client-js` typings which have been used more extensively at this point and were built with the benefits of modern typescript from the beginning.

Aside from that I no longer need this PR to acheive my original aims of using the auto-generated typings since I was able to avoid hitting the bug whilst also improving typings